rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -523,6 +523,45 @@ where
 		}))
 	}
 
+	pub fn light_cull(&self, txq: Arc<RwLock<TransactionQueue>>) -> impl Future <Item = (), Error = Error> + Send {
+		let senders = txq.read().queued_senders();
+		if senders.is_empty() {
+			return Either::B(future::err(errors::no_light_transactions()));
+		}
+
+		let sync = self.sync.clone();
+		let on_demand = self.on_demand.clone();
+		let best_header = self.client.best_block_header();
+		let start_nonce = self.client.engine().account_start_nonce(best_header.number());
+
+		let account_request = sync.with_context(move |ctx| {
+			// fetch the nonce of each sender in the queue.
+			let nonce_reqs = senders.iter()
+				.map(|&address| request::Account { header: best_header.clone().into(), address })
+				.collect::<Vec<_>>();
+
+			// when they come in, update each sender to the new nonce.
+			on_demand.request(ctx, nonce_reqs)
+				.expect("No back-references; therefore all back-references are valid; qed")
```

Comment:
I don't understand this, can you explain it?

Reply:
It is a `OnDemand` thing where you can make a `Request` that depends on `back-reference` which is a way to make request until the header is resolved

But the request doesn't have a back-reference because the header is known